### PR TITLE
WIP Paint setup/cleanup

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/drawing/supports.c [hunk 1]
```diff
@@ -231,9 +224,10 @@ bool wooden_a_supports_paint_setup(int supportType, int special, int height, uin
...
 			imageId |= imageColourFlags;
 
-			RCT2_GLOBAL(0x009DEA52, uint16) = 0;
-			RCT2_GLOBAL(0x009DEA54, uint16) = 0;
-			RCT2_GLOBAL(0x009DEA56, uint16) = z + 2;
+			uint16 word_9DEA52 = 0;
+			uint16 word_9DEA54 = 0;
+			uint16 word_9DEA56 = z + 2;
+			// Todo: check whether this function call is correct
 			sub_98196C(imageId, 0, 0, 32, 32, 11, z, rotation);
```
Comment (marijnvdwerf) [inline src/drawing/supports.c]: @duncanspumpkin I think this function call is wrong. Can you check?

File: src/interface/viewport.c [hunk 2]
```diff
@@ -920,10 +927,12 @@ int sub_98196C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, si
...
  * @param height (edx)
  * @param rotation
- * @return ??
+ * @param word_52 (0x009DEA52)
+ * @param word_54 (0x009DEA54)
+ * @param word_56 (0x009DEA56)
  */
-int sub_98197C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation){
-	int ebp = z_offset + RCT2_GLOBAL(0x9DEA56, sint16);
+int sub_98197C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation, sint16 s_unk_x, sint16 s_unk_y, sint16 word_56) {
```
Comment (marijnvdwerf) [inline src/interface/viewport.c]: @IntelOrca @duncanspumpkin Pretty much all assignments are done with `uint16`, but reading is being done with `sint16`. Which one should it be?

File: src/interface/viewport.h [hunk 3]
```diff
@@ -132,9 +132,9 @@ void sub_688485();
 void sub_688217();
 
 int sub_98196C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation);
-int sub_98197C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation);
-int sub_98198C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation);
-int sub_98199C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation);
+int sub_98197C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation, sint16 s_unk_x, sint16 s_unk_y, sint16 word_56);
+int sub_98198C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation, uint16 word_52, uint16 word_54, uint16 word_56);
+int sub_98199C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, sint16 length_y, sint8 z_offset, int height, uint32 rotation, uint16 word_52, uint16 word_54, uint16 word_56);
```
Comment (marijnvdwerf) [inline src/interface/viewport.h]: I still need to move `rotation` to the last argument position, to match the assembly assignment order.

File: src/drawing/supports.c (hunk 1 above)
Comment (duncanspumpkin) [inline src/drawing/supports.c]: Should be 98197C function call

**Comment by duncanspumpkin** [comment]
Can you label 0x009DEA52 and such in addresss.h

File: src/interface/viewport.c [hunk 4]
```diff
@@ -1703,9 +1694,9 @@ void viewport_banner_paint_setup(uint8 direction, int height, rct_map_element* m
 	direction += map_element->properties.banner.position;
 	direction &= 3;
 
-	RCT2_GLOBAL(0x9DEA56, uint16_t) = height + 2;
-
-	RCT2_GLOBAL(0x9DEA52, uint32_t) = RCT2_ADDRESS(0x98D884, uint32)[direction * 2];
+	boundBoxOffsetX = RCT2_ADDRESS(0x98D884, uint16)[direction * 2];
```
Comment (duncanspumpkin) [inline src/interface/viewport.c]: direction * 4 as we are now referencing a uint16 instead of uint32. This was the source of #3345.

**Pull request merged**